plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2488,8 +2488,20 @@ protected ConsoleDef createConsoleDef() {
         return new ConsoleDef(PTY, null, null, (short)0);
     }
 
-    protected VideoDef createVideoDef() {
-        return new VideoDef(_videoHw, _videoRam);
+    protected VideoDef createVideoDef(VirtualMachineTO vmTO) {
+        Map<String, String> details = vmTO.getDetails();
+        String videoHw = _videoHw;
+        int videoRam = _videoRam;
+        if (details != null) {
+            if (details.containsKey(VideoDef.VIDEO_MODEL)) {
```

Comment:
```suggestion
            if (details.containsKey(VmDetailConstants.VIDEO_HARDWARE)) {
```

can use the same constants defined in VmDetailConstants.